metrics/prometheus.go

```diff
@@ -571,3 +581,11 @@ func specMemoryValue(v uint64) float64 {
 	}
 	return float64(v)
 }
+
+var invalidLabelCharRE = regexp.MustCompile(`[^a-zA-Z0-9_]`)
+
+// sanitizeLabelName replaces anything that doesn't match
+// client_label.LabelNameRE with an underscore.
+func sanitizeLabelName(name string) string {
```

Comment:
We should sanitize rather than drop. This is normal practice for prometheus metrics.